File: 9_morning/20_05_11_angular/angular/src/app/service/contact.service.ts
import {Injectable} from '@angular/core';
import {Contact} from '../model/contact';

@Injectable()
export class ContactService {

  static currentId = 0;
  private readonly contacts: Contact[];

  constructor() {
    this.contacts = [];
  }

  add(contact: Contact): void {
    contact.id = ++ContactService.currentId;
    this.contacts.push(contact);
  }

  getAll(): Contact[] {
    console.log(this.contacts);
    return this.contacts;
  }
}

File: 11/20_10_12_contacts/src/main/java/de/telran/contacts/service/ContactService.java
package de.telran.contacts.service;

import de.telran.contacts.model.Contact;
import de.telran.contacts.repository.IContactRepo;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ContactService {

    final private IContactRepo contactRepo;

    public ContactService(IContactRepo contactRepo) {
        this.contactRepo = contactRepo;
    }

    public void add(Contact contact) {
        contactRepo.add(contact);
    }

    public List<Contact> getAll() {
        return contactRepo.getAll();
    }

    public void remove(int id) {
        contactRepo.remove(id);
    }

    public Contact get(int id) {
        return contactRepo.getById(id);
    }

    public void edit(Contact contact) {
        contactRepo.edit(contact);
    }
}
